### PR TITLE
NF: Applying Decorators in Module (Direction)

### PR DESCRIPTION
**Pull request by deka27**
## This PR is to apply keyword decorator to functions in direction module.

This PR is dependent on the main PR, which should be merged first.

## Main PR:
- #3239

### CODE REVIEW

**Comment by skoudoro** [comment]
Can you rebase and fix the format since #3239 is merged ? Thanks!

File: dipy/direction/ptt_direction_getter.pyx [hunk 1]
```diff
@@ -20,6 +20,7 @@ cimport numpy as cnp
 from libc.math cimport M_PI, pow, sin, cos, fabs
 from libc.stdlib cimport malloc, free
 
+from dipy.testing.decorators import warning_for_keywords
```
Comment (skoudoro) [inline dipy/direction/ptt_direction_getter.pyx:23]: no used so not needed I suppose

File: dipy/direction/probabilistic_direction_getter.pyx [hunk 2]
```diff
@@ -28,7 +29,7 @@ cdef class ProbabilisticDirectionGetter(PmfGenDirectionGetter):
     set to 0 and the result is normalized.
     """
 
-    def __init__(self, pmf_gen, max_angle, sphere, pmf_threshold=.1, **kwargs):
+    def __init__(self, pmf_gen, max_angle, sphere, pmf_threshold=.1 , **kwargs):
```
Comment (skoudoro) [inline dipy/direction/probabilistic_direction_getter.pyx:32]: missing `*`before  `pmf_threshold=.1 `

**Review by skoudoro — COMMENTED** [review]
Hi @deka27, See below my comments

File: dipy/direction/probabilistic_direction_getter.pyx (hunk 2 above)
Comment (deka27) [reply dipy/direction/probabilistic_direction_getter.pyx:32]: Adding the `*` here will provide that error we spoke about earlier. There is some particular conflict with `__init__` method for class. It passes an extra argument by itself.

File: dipy/direction/ptt_direction_getter.pyx (hunk 1 above)
Comment (deka27) [reply dipy/direction/ptt_direction_getter.pyx:23]: Done.

File: dipy/direction/closest_peak_direction_getter.pyx [hunk 3]
```diff
@@ -65,7 +66,7 @@ cdef int closest_peak(cnp.ndarray[cnp.float_t, ndim=2] peak_dirs,
 cdef class BasePmfDirectionGetter(DirectionGetter):
     """A base class for dynamic direction getters"""
 
-    def __init__(self, pmf_gen, max_angle, sphere, pmf_threshold=.1, **kwargs):
+    def __init__(self, pmf_gen, max_angle, sphere, *, pmf_threshold=.1, **kwargs):
```
Comment (skoudoro) [inline dipy/direction/closest_peak_direction_getter.pyx:69]: missing the decorator

File: dipy/direction/peaks.py [hunk 4]
```diff
@@ -661,6 +672,7 @@ def reshape_peaks_for_visualization(peaks):
     return peaks.reshape(np.append(peaks.shape[:-2], -1)).astype("float32")
 
 
+@warning_for_keywords()
```
Comment (skoudoro) [inline dipy/direction/peaks.py:675]: not needed since `*` was there from the beginning

File: dipy/direction/probabilistic_direction_getter.pyx [hunk 5]
```diff
@@ -28,7 +29,7 @@ cdef class ProbabilisticDirectionGetter(PmfGenDirectionGetter):
     set to 0 and the result is normalized.
     """
 
-    def __init__(self, pmf_gen, max_angle, sphere, pmf_threshold=.1, **kwargs):
+    def __init__(self, pmf_gen, max_angle, sphere, *, pmf_threshold=.1 , **kwargs):
```
Comment (skoudoro) [inline dipy/direction/probabilistic_direction_getter.pyx:32]: missing the decorator

**Review by skoudoro — APPROVED** [review]
Hi @deka27, 

Can you finalize this PR please? See below the last comment. Thank you

File: dipy/direction/peaks.py (hunk 4 above)
Comment (skoudoro) [reply dipy/direction/peaks.py:673]: you need to remove the decorator in the line above. This function always existed with * so we do not need to warn the user

File: dipy/direction/probabilistic_direction_getter.pyx (hunk 2 above)
Comment (skoudoro) [reply dipy/direction/probabilistic_direction_getter.pyx:30]: please, remove the extra space after `pmf_threshold=.1` to be back as what it was before

**Review by skoudoro — APPROVED** [review]
All good, thank you for this @deka27, merging !

**Pull request merged**